src/generic/svcb.h

```diff
@@ -366,6 +366,51 @@ static int32_t parse_unknown(
   return 0;
 }
 
+nonnull_all
+static int32_t parse_tls_supported_groups(
+  parser_t *parser,
+  const type_info_t *type,
+  const rdata_info_t *field,
+  uint16_t key,
+  const svc_param_info_t *param,
+  rdata_t *rdata,
+  const token_t *token)
+{
+  const char *t = token->data, *te = token->data + token->length;
+  const uint8_t *rdata_start = rdata->octets;
+
+  (void)field;
+  (void)key;
+  (void)param;
+
+  do {
+    uint64_t number = 0;
+    for (;; t++) {
+      const uint64_t digit = (uint8_t)*t - '0';
+      if (digit > 9)
+        break;
+      number = number * 10 + digit;
+    }
+
+    uint16_t group = (uint16_t)number;
+    group = htobe16(group);
+    memcpy(rdata->octets, &group, 2);
+    rdata->octets += 2;
+    if (number > 65535)
+      SYNTAX_ERROR(parser, "Invalid tls-supported-group in %s", NAME(type));
+
+    const uint8_t *g;
+    for (g = rdata_start; g < rdata->octets - 2; g += 2) {
+      if (memcmp(g, &group, 2) == 0)
+        SEMANTIC_ERROR(parser, "Duplicate group in tls-supported-groups in %s", NAME(type));
+    }
+  } while (t < te && *t++ == ',');
+
+  if (t != te || rdata->octets > rdata->limit)
```

Comment:
If the rdata limit is less than 128k, then the buffer could be smaller than the input. Checking here is too late, since the overwrites happened in the loop previously. Perhaps jump out of that loop if `rdata->octets+2 > rdata->limit`?

Reply:
I noticed the pattern of post-write checking in other parts of the source, but indeed for much smaller potential writes. The leeway is probably not 128k, so resolved in commit c1acc4cae to conservatively not expect any leeway